### PR TITLE
use next instead of forloop & replace set with comprehension

### PR DESCRIPTION
**Pull request by Mng-dev-ai**
This PR adds a slight speed improvement by using next instead of for loop and using {} instead of set() which should be faster than before

### CODE REVIEW

**Comment by sjsadowski** [comment]
Please add a description outlining both the value and purpose of this change.

**Comment by Mng-dev-ai** [comment]
@sjsadowski  Added

File: sanic/blueprints.py [hunk 1]
```diff
@@ -449,16 +449,11 @@ def event(self, event: str, timeout: Optional[Union[int, float]] = None):
 
     @staticmethod
     def _extract_value(*values):
-        value = values[-1]
-        for v in values:
-            if v is not None:
-                value = v
-                break
-        return value
+        return next((v for v in values if v is not None), values[-1])
```
Comment (sjsadowski) [inline sanic/blueprints.py:452]: Is there a reason for using a generator instead of a list comprehension here?

**Review by sjsadowski — CHANGES_REQUESTED** [review]

File: sanic/blueprints.py (hunk 1 above)
Comment (Mng-dev-ai) [reply sanic/blueprints.py:452]: I think generators are better to be used when it's possible as it uses less memory since it doesn't build the whole list at once

Comment (sjsadowski) [inline sanic/blueprints.py:452]: @ahopkins is this to really just return the first value found that is not None? Just want to make sure I understand what you were after here. I'm okay with switching to the generator as it's a parity change, but I want to verify that's what you were really after.

Comment (Mng-dev-ai) [reply sanic/blueprints.py:452]: yea, it returns the first value found that is not None

Comment (sjsadowski) [reply sanic/blueprints.py:452]: @Mng-dev-ai yes, but I want to make sure that you didn't uncover a bug, @ahopkins will probably look tomorrow.

Comment (ahopkins) [reply sanic/blueprints.py:452]: The logic seems fine. We probably could improve DRY a little by using `_extract_value` with the `version_prefix` and just abstract the conditional a bit since they are essentially doing the same thing, and it would keep the priority setting using a consistent methodology.

However, TBH, I am not sure how this is an improvement and what this change adds. Even running 10M iterations over it, I only had to run my benchmarks twice to get inconsistent results. I am not sure that I would agree that a generator adds any performance here.
image

We are only looping through 2 or three items, not a very large list. And, the values already exist in memory so using a generator does not reduce the footprint. Am I missing something else?

File: sanic/blueprints.py [hunk 2]
```diff
@@ -449,16 +449,11 @@ def event(self, event: str, timeout: Optional[Union[int, float]] = None):
...
 
     @staticmethod
     def register_futures(
         apps: Set[Sanic], bp: Blueprint, futures: Sequence[Tuple[Any, ...]]
     ):
         for app in apps:
-            app._future_registry.update(set((bp, item) for item in futures))
+            app._future_registry.update({(bp, item) for item in futures})
```
Comment (ahopkins) [inline sanic/blueprints.py:459]: What is the purpose of this change?

File: sanic/blueprints.py (hunk 1 above)
Comment (ahopkins) [reply sanic/blueprints.py:452]: I am not sure that I agree with this logic here. Arguments are passed as an existing object (a tuple). They already are existing in memory in an iterable. Where are we rebuilding the list that we are incurring a memory overhead?

Comment (Mng-dev-ai) [reply sanic/blueprints.py:452]: yea, in this case here there's no memory overhead the main reason here for using generators is to use next instead of forloop

Comment (Mng-dev-ai) [reply sanic/blueprints.py:452]: Didn't know that the array has only a few items so as u said the difference won't be noticed

File: sanic/blueprints.py (hunk 2 above)
Comment (Mng-dev-ai) [reply sanic/blueprints.py:459]: no need to use set() here as it's slower than just {}

**Comment by ahopkins** [comment]
Sorry @Mng-dev-ai. I am closing this for now as I do not see how this improves the codebase. Feel free to convince me otherwise if you think I am wrong.

1. Using the generator instead of the forloop does not add anything.
2. Using a set comprehension instead of `set` also does not add anything in this context.

Both of these items run only at startup. While it is a stated goal to increase performance of the project, we are mainly focused on the request/response cycle and not just marginal startup costs.
